woocommerce/woocommerce-gutenberg-products-block
# Experiment: Handling API fetch errors

This is an experiment into handling API fetch errors. The approach I've take is to store the API error in `state` and then render a placeholder which shows the error and allows you to retry.

Ref: #380

If we did go this route, other blocks would need to be updated to include this handling.

### Screenshots

This shows how it gets rendered:
![2019-07-03 15 28 46](https://user-images.githubusercontent.com/90977/60600222-af709100-9da7-11e9-98b8-62953541d8f5.gif)

You can retry as much as you want. if you want console you'll see the error 500 from the API.

### How to test the changes in this Pull Request:

1. Build
2. Add a Featured Product Block
3. Select a product - API error (hardcoded) will trigger.

Looking for feedback in terms of dev and design here. Since this is API error handling it **should not** be too common an occurrence as something would seriously be wrong, so is mostly here for safety.

Also, if I've missed pre-existing patterns for this I apologise and hope you'll tell me :) cc @ryelle who logged the original issue. Is this inline with your ideas?
